libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/VerifiableCredential.java

```diff
@@ -35,6 +40,14 @@ private VerifiableCredential(String userId, String criId, SignedJWT signedJwt, I
             throw new CredentialParseException(
                     "Failed to get jwt claims to construct verifiable credential", e);
         }
+
+        try {
+            VerifiableCredentialParser.parseCredential(claimsSet);
```

Comment:
We're going to have a name clash at some point, between the model `VerifiableCredential` types and this class. Either we'll just have to live with it (and specify if we need to handle both) or rename this class (which will be very noisy)